packages/opentelemetry-tracing/src/Span.ts

```diff
@@ -194,6 +190,13 @@ export class Span implements api.Span, ReadableSpan {
     return this._ended;
   }
 
+  get resource(): Resource {
```

Comment:
If the promise is resolved (after _resolveResources completes), then this will act like a regular sync resource to satisfy the ReadableSpan interface.

Reply:
Imagine the following situation
1. Resource is provided as promise
2. Start span
3. Something is checking for resource, then just started span receives a resource which is `Resource.createTelemetrySDKResource()`
4. Resource that was provided as a promise has been just resolved
5. Span ended - export span
6. Span is being exported with wrong resource instead of the one provided from promise

Another situation
1. Resource is provided as promise
2. Start span
3. Span ended - export span
4. Span is being exported with wrong resource instead of the one provided from promise
5. Resource that was provided as a promise has been just resolved - but previously created span has been already exported with wrong resource

Reply:
>Something is checking for resource

Checking for resource isn't even a part of the public API until we get a "Readable Span", which is not the same as a span. It might be easier if we just send a plain object instead of the actual Span with a getter?

>Span is being exported with wrong resource instead of the one provided from promise

No, the export will still have the correct resource as the span will not be sent to the exporter until the resource is resolved, and getters don't cache. Try it if you don't believe me.

>Another situation

>Span is being exported with wrong resource instead of the one provided from promise

Why? The span isn't actually exported until the resource resolves.

Reply:
?

Reply:
I wrote you a message on gitter 4 days ago, please read :)

Reply:
D: ok thanks

Too many communication channels to keep track of